engines/web/app/controllers/web/application_controller.rb

```diff
@@ -0,0 +1,5 @@
+module Web
+  class ApplicationController < ActionController::Base
+    protect_from_forgery with: :exception
```

Comment:
Why does this need to be overridden instead of using the regular ApplicationController value?

Reply:
Ok, I guess I'm a bit confused about how this and the main ApplicationController will work together.

Reply:
the easiest way to think about an engine is to think of it as a separate app (from the guide "Engines can be considered miniature applications").
we can get to the engine code through requiring its modules and classes but the main way to get to engine will be through its routes (there's a new entry on the main app routes to include all the routes of this engine), so this is a different app and so we extend ActionController::Base.

Actually, this is a good example of where we will get the opportunity to better organize our code. If you look at the main ApplicationController you see it's contains code relate to  embedded_shopfront code, this code can be moved to this Web module ApplicationController and removed from the main app and thus, cleaning up this code from the other spaces, like the unrelated backoffice code.

the reference for engines is this: https://guides.rubyonrails.org/engines.html